### ISSUE TITLE
Deno.exit() in switch statement yielts no-fallthrough warning without a break keyword

### ISSUE THREAD

**Issue by noxifoxi** [denoland/deno_lint#1042]
### Lint Name

no-fallthrough

### Code Snippet

```ts
switch (input) {
	case "1":
		break;
	case "x":
		Deno.exit();

	default:
		break;
}
```

### Expected Result
No warnings about case fallthroughs 

### Actual Result
Warns about a fallthrough

### Additional Info
deno-lint is not happy until I add `break;` after `Deno.exit();`, although it's unreachable:
![image](https://user-images.githubusercontent.com/293639/167509959-be256cc0-31e6-4191-9b62-a653d6474b08.png)

deno-ts(7027) knows and puts a notice about unreachable code:
![image](https://user-images.githubusercontent.com/293639/167509893-36d4e386-640c-44aa-abcb-f0b7afaa0a63.png)

### Version
VS-Code Extension: v3.12.0

deno 1.21.2 (release, x86_64-pc-windows-msvc)
v8 10.0.139.17
typescript 4.6.2

**Comment by bartlomieju** [comment]
This is somewhat of a design limitation at this point... `deno lint` is not really aware that `Deno.exit()` has a return type of `never`. I wonder if we could handle that with @kdy1 control flow analysis? 

CC @dsherret

**Comment by kdy1** [comment]
Using `never` type requires a good enough type checker

We can use a 'known list' of expression with never type alternatively

**Comment by bartlomieju** [comment]
> Using `never` type requires a good enough type checker
> 
> We can use a 'known list' of expression with never type alternatively

That's a good point, it's much simpler solution than incorporating type checking and should be good enough in this case.
